screens/Guest/GuestViolation.js

```diff
@@ -0,0 +1,466 @@
+  const [role, setRole] = useState('');
+  const [guestId, setGuestId] = useState(null);
+  const navigation = useNavigation();
+
+  /**
+   * Initializes the component by checking the token, role, and guestId in AsyncStorage.
+   * If valid, it fetches violations for the given guestId.
+   * If any of these values are missing or invalid, it logs out the user.
+   */
+  useEffect(() => {
+    const initialize = async () => {
+      try {
+        const token = await AsyncStorage.getItem('token');
+        const storedRole = await AsyncStorage.getItem('role');
+        const storedGuestId = await AsyncStorage.getItem('guestId');
+
+        if (!token || !storedRole || !storedGuestId) {
+          Alert.alert('Error', 'Session expired. Please log in again.');
+          handleLogout();
+          return;
+        }
+
+        setGuestId(storedGuestId);
+        fetchViolations(storedGuestId);
+      } catch (err) {
+        console.error('Initialization error:', err);
+        Alert.alert('Error', 'An error occurred during initialization.');
+        setLoading(false);
+      }
+    };
+
+    initialize();
+  }, []);
+
+  /**
+   * Fetches violations for the guest's beneficiaries and their students.
+   * It makes multiple API requests to gather violation data for each student and stores them in the state.
+   * 
+   * @param {string} guestId - The guest ID used to fetch beneficiaries and their violations.
+   * 
+   * 
+   * Note: Change the IP address in the axios URL to match your backend server's IP address and port.
+   */
+  const fetchViolations = async (guestId) => {
+    setLoading(true);
+    setError(null);
+
+    try {
+      const token = await AsyncStorage.getItem('token');
+      const response = await fetch(`http://192.168.1.8:8080/guest/${guestId}/Beneficiaries`, {
+        headers: {
+          'Content-Type': 'application/json',
+          Authorization: `Bearer ${token}`,
+        },
+      });
+
+      const responseBody = await response.text();
+
+      if (!response.ok) {
+        throw new Error(`Failed to fetch beneficiaries. HTTP Status: ${response.status}`);
+      }
+
+      const beneficiariesData = JSON.parse(responseBody);
+
+      if (!Array.isArray(beneficiariesData)) {
+        throw new Error('The response is not an array. Check the API response.');
+      }
+
+      const fetchedViolations = await Promise.all(
+        beneficiariesData.flatMap((beneficiary) =>
+          beneficiary.beneficiary.map(async (student) => {
+            try {
+              const studentNumber = student.studentNumber;
+              const violationResponse = await fetch(
+                `http://192.168.1.8:8080/violation/studentNumber/${studentNumber}`,
+                {
+                  headers: {
+                    'Content-Type': 'application/json',
+                    Authorization: `Bearer ${token}`,
+                  },
+                }
+              );
+
+              if (!violationResponse.ok) {
```

Comment:
This is redundant, you already have a catch error('Failed to fetch violations for ${studentNumber}') on line 123